Explorer/Assets/DCL/Multiplayer/SDK/Systems/WriteAvatarEmoteCommandSystem.cs

```diff
@@ -0,0 +1,91 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using CRDT;
+using CrdtEcsBridge.ECSToCRDTWriter;
+using DCL.Diagnostics;
+using DCL.ECSComponents;
+using DCL.Multiplayer.SDK.Components;
+using DCL.Optimization.Pools;
+using ECS.Abstract;
+using ECS.LifeCycle.Components;
+using SceneRunner.Scene;
+
+namespace DCL.Multiplayer.SDK.Systems
+{
+    [UpdateInGroup(typeof(PresentationSystemGroup))]
+    [UpdateAfter(typeof(PlayerComponentsHandlerSystem))]
+    [LogCategory(ReportCategory.PLAYER_AVATAR_EMOTE_COMMAND)]
+    public partial class WriteAvatarEmoteCommandSystem : BaseUnityLoopSystem
+    {
+        private readonly ISceneStateProvider sceneStateProvider;
+        private readonly IECSToCRDTWriter ecsToCRDTWriter;
+        private readonly IComponentPool<PBAvatarEmoteCommand> componentPool;
+
+        public WriteAvatarEmoteCommandSystem(World world, IECSToCRDTWriter ecsToCRDTWriter, IComponentPool<PBAvatarEmoteCommand> componentPool, ISceneStateProvider sceneStateProvider) : base(world)
+        {
+            this.ecsToCRDTWriter = ecsToCRDTWriter;
+            this.componentPool = componentPool;
+            this.sceneStateProvider = sceneStateProvider;
+        }
+
+        protected override void Update(float t)
+        {
+            HandleComponentRemovalQuery(World);
+            UpdateAvatarEmoteCommandQuery(World);
+            CreateAvatarEmoteCommandQuery(World);
+        }
+
+        [Query]
+        [None(typeof(PBAvatarEmoteCommand), typeof(DeleteEntityIntention))]
+        private void CreateAvatarEmoteCommand(in Entity entity, ref PlayerSDKDataComponent playerSDKDataComponent)
+        {
+            if (playerSDKDataComponent.PlayingEmote == null) return;
+
+            PBAvatarEmoteCommand pbComponent = componentPool.Get();
```

Comment:
I think it's a bit confusing that we add the result component (the only purpose of which is to propagate data to SDK) to ECS on our side.
Can you review the following option:
1. Instead of `PlayerSDKDataComponent` introduce multiple components with single responsibility
2. Write into the component that is solely dedicated to the emote command all the data you need internally
3. This will let you get rid of the pooling of this extra command

Reply:
I managed to get rid of the pooled components by relying on the `IsDirty` prop of other components.